etc/kayobe/ansible/growroot.yml

```diff
@@ -0,0 +1,62 @@
+---
+# Custom playbook to grow the partition and LVM PV of the root VG. This allows
+# for expansion of LVs in that VG. This may be used as a pre-overcloud host
+# configure hook.
+#
+# Variables:
+# * growroot_group: Host pattern against which to target the playbook. Default
+#                   is 'overcloud'.
+# * growroot_vg: Name of the VG containing the PV to grow. Default is 'rootvg'.
+
+- name: Grow root PV
+  hosts: "{{ growroot_group | default('overcloud') }}"
+  # Avoid using facts because this may be used as a pre overcloud host
+  # configure hook, and we don't want to populate the fact cache (if one is in
+  # use) with the bootstrap user's context.
+  gather_facts: false
+  tags:
+    - growroot
+  vars:
+    ansible_user: "{{ bootstrap_user }}"
+    # We can't assume that a virtualenv exists at this point, so use the system
+    # python interpreter.
+    ansible_python_interpreter: /usr/bin/python3
+    # Work around no known_hosts entry on first boot.
+    ansible_ssh_common_args: "-o StrictHostKeyChecking=no"
+    # Name of the LVM VG containing the root PV.
+    growroot_vg: "rootvg"
+
+  tasks:
+    - name: Ensure growpart is installed
+      package:
+        name: "{% if os_distribution in ['centos', 'rocky'] %}cloud-utils-growpart{% else %}cloud-guest-utils{% endif %}"
```

Comment:
This uses facts but the playbook doesn't collect them:

```
- name: Grow root PV
  hosts: "{{ growroot_group | default('overcloud') }}"
  # Avoid using facts because this may be used as a pre overcloud host
  # configure hook, and we don't want to populate the fact cache (if one is in
  # use) with the bootstrap user's context.
  gather_facts: false
```

Reply:
I seems you can use `meta: clear_facts` to clear the fact cache, but I guess clearing the cache is a nasty side effect!

Reply:
Possibly we can use `setup` with `gather_subset` to only collect the ones we need?

Reply:
Good catch. I'll use facts, with a fallback to `os_distribution`.